docs/source/reference/udf_decorators.rst

```diff
@@ -0,0 +1,153 @@
+User-Defined Functions
+======================
+
+This section provides an overview of how you can create and use a custom user-defined function (UDF) in your queries. For example, you could write an UDF that wraps around a PyTorch model.
+
+
+Part 1: Writing a custom UDF
+-----
+.. admonition:: Illustrative UDF implementation
+
+    During each step, use `this UDF implementation <https://github.com/georgia-tech-db/eva/blob/master/eva/udfs/decorators/yolo_object_detection_decorators.py>`_  as a reference.
+
+1. Create a new file under `udfs/` folder and give it a descriptive name. eg: `yolo_object_detection.py`. 
+
+.. note::
+
+    UDFs packaged along with EVA are located inside the `udfs <https://github.com/georgia-tech-db/eva/tree/master/eva/udfs>`_ folder.
+
+2. Create a Python class that inherits from `PytorchClassifierAbstractUDF`.
+
+* The `PytorchClassifierAbstractUDF` is a parent class that defines and implements standard methods for model inference.
```

Comment:
Why are we introducing `PytorchClassifierAbstractUDF` upfront? How about we break it into using `AbstractUDF` etc.? And it is not a classification job so we should also change it in the system. 
We had a bad design earlier, let us carefully fix most of it now.

Reply:
Do we want to do this in this PR or a different one?